cmd/metal-api/internal/headscale/client.go

```diff
@@ -119,6 +119,30 @@ func (h *HeadscaleClient) CreatePreAuthKey(namespace string, expiration time.Tim
 	return resp.PreAuthKey.Key, nil
 }
 
+func (h *HeadscaleClient) DescribeMachine(machineid, projectID string) (connected bool, err error) {
+	req := &headscalev1.ListMachinesRequest{
+		Namespace: projectID,
+	}
+	resp, err := h.client.ListMachines(h.ctx, req)
+	if err != nil || resp == nil {
+		return false, fmt.Errorf("failed to list machines: %w", err)
+	}
+
+	for _, m := range resp.Machines {
+		if m.Name == machineid {
```

Comment:
i thought a check for the tag containing the machineid is now possible, why is the name now equal to machineid ?

Reply:
https://github.com/metal-stack/metal-api/pull/336#discussion_r975745133, unfortunately it isn't possible if ACL policy isn't set.